WS2012R2/lisa/remote-scripts/ica/Hyperv_Daemons_Files_Status.sh

```diff
@@ -0,0 +1,219 @@
+
+CheckDaemonsFilesRHEL7()
+{
+  dameonFile=`ls /usr/lib/systemd/system | grep -i $1`
+  dameonFile2=`ls /etc/systemd/system/multi-user.target.wants | grep -i $1`
+  if [[ "$dameonFile" != $1 ]] || [[ "$dameonFile2" != $1 ]] ; then
+    LogMsg "ERROR: $1 is not in /usr/lib/systemd/system or /etc/systemd/system/multi-user.target.wants , test failed"
+    UpdateSummary "ERROR: $1 is not in /usr/lib/systemd/system or /etc/systemd/system/multi-user.target.wants , test failed"
+    UpdateTestState $ICA_TESTFAILED
+    exit 1
+  fi
+
+  dameonPreset=`cat /lib/systemd/system-preset/90-default.preset | grep -i $1`
+  if [ "$dameonPreset" != "enable $1" ]; then
+    LogMsg "ERROR: $1 is not in 90-default.preset, test failed"
+    UpdateSummary "ERROR: $1 is not in 90-default.preset, test failed"
+    UpdateTestState $ICA_TESTFAILED
+    exit 1
+  fi
+}
+
+#######################################################################
+# Check hyper-v daemons related file under default folder for rhel 6
+#######################################################################
+CheckDaemonsFilesRHEL6()
+{
+  dameonFile=`ls /etc/rc.d/init.d | grep -i $1`
+
+  if [[ "$dameonFile" != $1 ]] ; then
+    LogMsg "ERROR: $1 is not in /etc/rc.d/init.d , test failed"
+    UpdateSummary "ERROR: $1 is not in /etc/rc.d/init.d , test failed"
+    UpdateTestState $ICA_TESTFAILED
+    exit 1
+  fi
+}
+
+#######################################################################
+# Check hyper-v daemons service status is active for rhel7
+#######################################################################
+CheckDaemonsStatusRHEL7()
+{
+  dameonStatus=`systemctl is-active $1`
+  if [ $dameonStatus != "active" ]; then
+    LogMsg "ERROR: $1 is not in running state, test aborted"
+    UpdateSummary "ERROR: $1 is not in running state, test aborted"
+    UpdateTestState $ICA_TESTABORTED
+    exit 1
+
+  fi
+}
+
+#######################################################################
+# Check hyper-v daemons service status is active
+#######################################################################
+CheckDaemonsStatus()
+{
+  if [[ $(ps -ef | grep $1 | grep -v grep)  ]] || \
+       [[ $(ps -ef | grep $2 | grep -v grep) ]]; then
+      LogMsg "$1 Daemon is running"
+
+  else
+      LogMsg "ERROR: $1 Daemon not running, test aborted"
+      UpdateSummary "ERROR: $1 Daemon not running, test aborted"
+      UpdateTestState $ICA_TESTABORTED
+      exit 1
+  fi
+}
+
+#######################################################################
+# Main script body
+#######################################################################
+
+# Create the state.txt file so ICA knows we are running
+UpdateTestState $ICA_TESTRUNNING
+
+# Cleanup any old summary.log files
+if [ -e ~/summary.log ]; then
+    rm -rf ~/summary.log
+fi
+
+# Source the constants file
+if [ -e constants.sh ]; then
+    . constants.sh
+else
+    LogMsg "WARN: Unable to source the constants file."
+fi
+
```

Comment:
please avoid multiple blank lines in the code, this should be only 1 blank line between functions or lines of code.